crates/aranya-runtime/src/storage/linear/libc/tests.rs

```diff
@@ -25,6 +27,59 @@ impl StorageBackend for LinearBackend {
     }
 }
 
+struct Init(CommandId);
+
+impl Command for Init {
+    fn priority(&self) -> Priority {
+        Priority::Init
+    }
+
+    fn id(&self) -> CommandId {
+        self.0
+    }
+
+    fn parent(&self) -> Prior<Address> {
+        Prior::None
+    }
+
+    fn policy(&self) -> Option<&[u8]> {
+        None
+    }
+
+    fn bytes(&self) -> &[u8] {
+        &[0]
+    }
+}
+
+#[test]
+fn test_multiple_graph_ids() {
+    let tempdir = tempfile::tempdir().unwrap();
+    info!(path = ?tempdir.path(), "using tempdir");
+    let mut backend = LinearBackend { tempdir };
+    let mut provider = backend.provider(0);
+
+    let init_cmd1 = Init([1u8; 64].into());
+    let mut fp1 = provider.new_perspective(PolicyId::new(0));
+    let _ = fp1.add_command(&init_cmd1);
+    let _ = provider.new_storage(fp1).unwrap();
+
+    let init_cmd2 = Init([2u8; 64].into());
+    let mut fp2 = provider.new_perspective(PolicyId::new(1));
+    let _ = fp2.add_command(&init_cmd2);
+    let _ = provider.new_storage(fp2).unwrap();
+
+    let mut graph_ids = provider
+        .list_graph_ids()
+        .unwrap()
+        .filter_map(Result::ok)
```

Comment:
```suggestion
        .map(Result::unwrap)
```

We shouldn't see any Errs in this test